### PR TITLE
feat: Reexport non-deprecated dplyr functions

### LINKED ISSUE #144: Consider (re)exporting all {dplyr} functions (or depending on it)
For consistency with `dplyr` it would be nice if `as_tibble()` was available.

``` r
Sys.setenv(DUCKPLYR_METHODS_OVERWRITE = FALSE)
library(duckplyr, warn.conflicts = FALSE)
methods_overwrite()
#> ℹ Overwriting dplyr methods with duckplyr methods
as_tibble(mtcars)
#> Error in as_tibble(mtcars): could not find function "as_tibble"
    
methods_restore()
#> ℹ Restoring dplyr methods
library(dplyr, warn.conflicts = FALSE)
as_tibble(mtcars)    
#> # A tibble: 32 × 11
#>      mpg   cyl  disp    hp  drat    wt  qsec    vs    am  gear  carb
#>    <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl> <dbl>
#>  1  21       6  160    110  3.9   2.62  16.5     0     1     4     4
#>  2  21       6  160    110  3.9   2.88  17.0     0     1     4     4
#>  3  22.8     4  108     93  3.85  2.32  18.6     1     1     4     1
#>  4  21.4     6  258    110  3.08  3.22  19.4     1     0     3     1
#>  5  18.7     8  360    175  3.15  3.44  17.0     0     0     3     2
#>  6  18.1     6  225    105  2.76  3.46  20.2     1     0     3     1
#>  7  14.3     8  360    245  3.21  3.57  15.8     0     0     3     4
#>  8  24.4     4  147.    62  3.69  3.19  20       1     0     4     2
#>  9  22.8     4  141.    95  3.92  3.15  22.9     1     0     4     2
#> 10  19.2     6  168.   123  3.92  3.44  18.3     1     0     4     4
#> # ℹ 22 more rows
```

<sup>Created on 2024-04-17 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by krlmlr**
Closes #144.

### CODE REVIEW

**Comment by krlmlr** [comment]
@DavisVaughan: Can you please take a look? Anything else we want to reexport, any deprecated functions I've included by accident?

**Comment by krlmlr** [comment]
Please review only the first comment, I'll need to add reexport statements once we have agreed on the set of reexports.

**Comment by krlmlr** [comment]
Thanks, Davis, for the extensive review.

I went through the existing reexports and identified the following obvious candidates for removal:

- [ ] `do()`
- [ ] `dplyr_col_modify()`
- [ ] `dplyr_reconstruct()`
- [ ] `dplyr_row_slice()`
- [ ] `funs()`
- [ ] `same_src()`
- [ ] `vars()`

The following are experimental, not sure:

- [ ] `group_map()`
- [ ] `group_modify()`
- [ ] `group_walk()`

In addition, there is a slew of functions related to grouping. At this stage, duckplyr does not support `group_by()` . Should it? Removal of the reexports might depend on if we think this should be implemented. Is there a way to do rowwise operations with the new `.by` argument? Should this be mentioned in `?dplyr::dplyr_by` ?

- [ ] `group_by()`
- [ ] `group_data()`
- [ ] `group_indices()`
- [ ] `group_keys()`
- [ ] `group_rows()`
- [ ] `group_size()`
- [ ] `group_vars()`
- [ ] `groups()`
- [ ] `is_grouped_df()`
- [ ] `n_groups()`
- [ ] `rowwise()`

**Comment by DavisVaughan** [comment]
Remove those "obvious candidates" and anything related to `groups_*()` seems fine to me if duckplyr supports and prefers `.by`. Otherwise it is kind of confusing that `group_by()` "falls back" but `.by` doesn't.

For `rowwise()`, we don't yet support that through `.by`, but it would likely be a separate mutually exclusive argument like `.by_row = FALSE/TRUE` so we can keep `.by` "pure" tidyselect, see https://github.com/tidyverse/dplyr/issues/6660

**Review by DavisVaughan — APPROVED** [review]

File: R/duckplyr-package.R [hunk 1]
```diff
@@ -12,14 +12,45 @@
 ## usethis namespace: start
 #' @importFrom collections dict
 #' @importFrom collections queue
+#' @importFrom dplyr auto_copy
+#' @importFrom dplyr do
+#' @importFrom dplyr dplyr_col_modify
```
[the remark this answers is not in the document]
Comment (krlmlr) [reply R/duckplyr-package.R:17]: Yes, for dplyr's tests converted to duckplyr.

**Pull request merged**